### PR TITLE
add single table inheritance- add models for pieces, add delgates to gam...

### PR DESCRIPTION
**Pull request by bfhcodehacker**
...e model and add classes and scope to piece model

 -added files for the piece classes to app/model
 -added piece classes to piece.rb
 -added scope for pieces to piece.rb
 -added delegate to game.rb 
 -used self.inheritance_column=:piece_type instead of rails :type so I did not update databse

   I don't think this needs any testing, but would be happy to add if I'm wrong.

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+  scope :pawns,   -> { where(piece_type: 'Pawn') }
+  scope :rooks,   -> { where(piece_type: 'Rook') }
+  scope :bishops, -> { where(piece_type: 'Bishop') }
+  scope :knights, -> { where(piece_type: 'Knight') }
+  scope :kings,   -> { where(piece_type: 'King') }
+  scope :queens,  -> { where(piece_type: 'Queen') }
+ 
 end
+
+class Pawn < Piece; end
```
Comment (kenmazaika) [inline app/models/piece.rb]: These lines shouldn't be down here, as they're defined in their own files.

File: app/models/pawn.rb [hunk 2]
```diff
@@ -0,0 +1 @@
+class Pawn < Piece; end
```
Comment (kenmazaika) [inline app/models/pawn.rb]: i'd put this across multiple lines for each of the classes.

File: app/models/piece.rb [hunk 3]
```diff
@@ -1,6 +1,27 @@
...
+  def self.piece_types
+    %w(Pawn Rook Bishop Knight Queen King)
+  end
 
+  scope :pawns,   -> { where(piece_type: 'Pawn') }
+  scope :rooks,   -> { where(piece_type: 'Rook') }
+  scope :bishops, -> { where(piece_type: 'Bishop') }
+  scope :knights, -> { where(piece_type: 'Knight') }
+  scope :kings,   -> { where(piece_type: 'King') }
+  scope :queens,  -> { where(piece_type: 'Queen') }
```
Comment (kenmazaika) [inline app/models/piece.rb]: these scopes are really the only thin you added that could potentially be tested.

Also, on your next mentor session with Sean you might want to ask him how you could use meta-programming in ruby to make lines 10-15, 3 lines of code :)

**Comment by kenmazaika** [comment]
Seems ok to not have any tests for this IMO.  Good work, Brad! :)

**Comment by bfhcodehacker** [comment]
Thanks, Ken! I'm going to cancel this pull request, fix the changes you suggested and open a new one. I'll also check out the ruby monk metaprogramming and talk to Sean.

**Comment by kenmazaika** [comment]
You actually don't need to close the pull request and re-open it to update it.  If you push changes on the same branch the pull request automatically gets updated to include the changes.

Closing the pull request is a solid way to make sure somebody doesn't accidentally push the merge button on code you don't want to have in the master branch though.

**Comment by bfhcodehacker** [comment]
reopening and updating... good to know.

**Pull request merged**